File: lecture_05/primes.go
package main

import (
    "fmt"
    "math"
    "time"
	"os"
)

const NUM_PRIMES = 100_000

func isPrime(number int) bool {
    limit := int(math.Sqrt(float64(number))) + 1
    for d := 2; d < limit; d++ {
        if number % d == 0 {
            return false
        }
    }
    return true
}

func genPrimes(numPrimes int) {
    for v := 2; numPrimes > 0; v++ {
        if isPrime(v) {
            fmt.Println(v)
            numPrimes--
        }
    }
}

func main() {
    fmt.Fprintf(os.Stderr, "Generating %d prime numbers...\n", NUM_PRIMES)
    startTime := time.Now()
    genPrimes(NUM_PRIMES)
    duration := time.Since(startTime).Seconds()
    fmt.Fprintf(os.Stderr, "genPrimes(%d) took %.6f seconds",
	            NUM_PRIMES, duration)
}

File: lecture_02/Primes.java
import java.math.*;

class Primes {
    public static final long NUM_PRIMES = 100_000;

    public static boolean isPrime(long number) {
        var limit = (long)Math.sqrt(number) + 1;

        for (var d=2; d < limit; d++) {
            if (number % d == 0)
                return false;
        }

        return true;
    }

    public static void generate(long numPrimes) {
        
        for (var v=2; numPrimes > 0; v++) {
            if (isPrime(v)) {
                System.out.println(v);
                --numPrimes;
            }
        }

    }

    public static void main(String[] args) {
        System.err.printf("Generating %d prime numbers...\n", NUM_PRIMES);

        var start_time = System.currentTimeMillis();
        Primes.generate(NUM_PRIMES);    
        var duration = (System.currentTimeMillis() - start_time) / 1000.0;
        
        System.err.printf("Primes.generate(%d) took %f seconds.\n",
                                                    NUM_PRIMES, duration);
    }
}
